rescan.go

```diff
@@ -722,6 +745,58 @@ rescanLoop:
 					log.Warnf("Received unhandled block "+
 						"notification: %T", ntfn)
 				}
+
+			// Our retry signal has fired, so we'll attempt to
+			// refetch and notify the filter for our next queued
+			// block.
+			case <-blockRetrySignal:
+				blockRetrySignal = nil
+
+				// It's possible for the signal to fire without
+				// having any blocks to retry if they were
+				// disconnected right before.
+				retryBlock := blockRetryQueue.peek()
+				if retryBlock == nil {
+					continue rescanLoop
+				}
+
+				err := handleBlockConnected(retryBlock)
+				switch err {
+				case nil:
+					// We successfully notified the block
+					// this time, so we can remove it from
+					// our queue. If we have another block
+					// to retry, we'll set up another signal
+					// to allow for a disconnected
+					// notification to potentially arrive
+					// for it.
```

Comment:
> Is it likely that that disconnected notification arrives at this point?

"Likely" depends on the chain. This is more about _allowing_ a disconnected notification to be received so that we properly stop retrying a stale block.

> When we are rescanning and there are no retries, there also isn't such a delay?

Rescanning where?

Reply:
What I mean is that if there is a reorg and we need to fetch a series of new blocks and those blocks are fetched correctly without retries, we also don't delay in between.